src/helpers.c

```diff
@@ -939,3 +939,16 @@ bool string_in_list(char *string_to_check, struct list *list_to_check)
 
 	return false;
 }
+
+void print_progress(unsigned int count, unsigned int max)
+{
+	if (isatty(fileno(stdout))) {
+		printf("\r\t...%d%%", (int)(100 * ((float)count / (float)max)));
```

Comment:
Even though this `printf` is overwriting the line, due to the carriage return, it seems wasteful. For example, if there are 100,000 files to full_download, 100,000 prints are issued (followed by a flush).

Can this be made smarter to only print when the percentage value changes?